src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import org.apache.commons.lang.ObjectUtils;
+
 public class ArrayList<T> implements List<T> {
+    private static int dimension = 10;
+    private static final double INCREASING_COEFFICIENT = 1.5;
+    private int size;
+    private T[] array = (T[]) new Object[dimension];
+
     @Override
     public void add(T value) {
-
+        addable();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        addable();
+        addIndexInRange(index);
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexInRange(index);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexInRange(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        addable();
+        indexInRange(index);
+        final T removeElement = array[index];
+        System.arraycopy(array, index + 1, array, index, size - index);
+        array[size - 1] = null;
+        size--;
+        return removeElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (ObjectUtils.equals(element, array[i])) {
```

Comment:
It is better to use Objects.equals(element, array[i]) which is available in Java7 without using external lib